src/Servers/IIS/IIS/test/Common.FunctionalTests/CompressionTests.cs

```diff
@@ -1,64 +1,96 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System.Linq;
+using System;
+using System.Collections.Generic;
+using System.Net;
 using System.Net.Http;
 using System.Net.Http.Headers;
 using System.Threading.Tasks;
-using Microsoft.AspNetCore.Server.IIS.FunctionalTests.Utilities;
-using Microsoft.AspNetCore.Server.IntegrationTesting;
 using Microsoft.AspNetCore.Testing;
 using Xunit;
 
 namespace Microsoft.AspNetCore.Server.IIS.FunctionalTests
 {
-    [Collection(PublishedSitesCollection.Name)]
-    public class CompressionTests : IISFunctionalTestBase
+    [Collection(IISCompressionSiteCollection.Name)]
+    public class CompressionTests : FixtureLoggedTest
     {
-        public CompressionTests(PublishedSitesFixture fixture) : base(fixture)
+        private readonly IISCompressionSiteFixture _fixture;
+
+        public CompressionTests(IISCompressionSiteFixture fixture): base(fixture)
         {
+            _fixture = fixture;
         }
 
-        [ConditionalFact]
-        public async Task PassesThroughCompressionOutOfProcess()
+        [ConditionalTheory]
+        [RequiresIIS(IISCapability.DynamicCompression)]
+        [InlineData(true)]
```

Comment:
This is a strong personal preference, but these are much better authored as two separate tests. You could use a private method to share the implementation. Having separate tests makes it much more legible what the test is about, easier to individually skip \ quarantine, and easy to search for \ filter. Theory is much better suited to cover different data to the same algorithm, but is crummy when describing different code paths \ scenarios.